### PR TITLE
Add support for fetching multiple header values

### PR DESCRIPTION
**Pull request by vermagav**

### CODE REVIEW

File: runtime/server_header.go [hunk 1]
```diff
@@ -33,7 +33,14 @@ import (
...
 // Header defines methods on ServerHeaders
 type Header interface {
+	// Get mirrors the implementation of http.header and returns a single header value.
+	// When a key contains multiple values, -only- the first one is returned.
+	// Ref: https://golang.org/pkg/net/http/#Header.Get
 	Get(key string) (string, bool)
+	// Values mirrors the implementation of http.header and returns a slice of header values.
+	// When a key contains multiple values, the entire collection is returned.
+	// Ref: https://golang.org/pkg/net/http/#Header.Values
+	//Values(key string) ([]string, bool)
```
Comment (rpatali) [inline runtime/server_header.go:43]: Forgot to uncomment?

Comment (vermagav) [reply runtime/server_header.go]: Yep, that was from a stale commit. Updates pending.

File: runtime/server_header.go [hunk 2]
```diff
@@ -151,6 +169,19 @@ func (th ServerTChannelHeader) Get(key string) (string, bool) {
...
 
+// Values retrieves the entire collection of values stored on a given header.
+// Bool return value is used to distinguish between the presence of a
+// header with golang's zerovalue slice and the absence of the header.
+func (th ServerTChannelHeader) Values(key string) ([]string, bool) {
+	if value, ok := th.Get(key); ok {
+		// In the case of TChannel, ServerTChannelHeader does not support
+		//  multiple, disparate values so we defer to Get and package it
+		// in a slice to meet the interface's requirement.
+		return []string{value}, ok
```
Comment (tejaswiagarwal) [inline runtime/server_header.go:180]: Can we do this instead? 

`return th.Get(key)`

**Comment by ChuntaoLu** [comment]
The ci failure is due to format inconsistency on travis ci, safe to ignore.

**Pull request merged**